src/Spectre.Service.Tests/Loaders/DatasetLoaderTest.cs

```diff
@@ -0,0 +1,109 @@
+{
+    [TestFixture]
+    public class DatasetLoaderTest
+    {
+        private DataRootConfig _rootConfig;
+        private DatasetLoader _datasetLoader;
+        private MockFileSystem _mockFileSystem;
+
+        private readonly string _rootDir = @"C:\spectre_data";
+        private readonly string _localDir = "local";
+        private readonly string _remoteDir = "remote";
+
+        private readonly string _fileDir = $"{TestContext.CurrentContext.TestDirectory} + "
+                                           + @"\..\..\..\..\..\test_files";
+        [OneTimeSetUp]
+        public void SetUp()
+        {
+            DependencyResolver.AddModule(new MockModule());
+
+            var localDirFull = Path.Combine(_rootDir, _localDir);
+            var remoteDirFull = Path.Combine(_rootDir, _remoteDir);
+            var correctDataset = File.ReadAllText(Path.Combine(_fileDir, "small-test.txt"));
+
+            _mockFileSystem = DependencyResolver.GetService(typeof(IFileSystem)) as MockFileSystem;
+
+            _mockFileSystem.AddFile(Path.Combine(localDirFull, "local_correct.txt"), new MockFileData(correctDataset));
+            _mockFileSystem.AddFile(Path.Combine(remoteDirFull, "remote_correct.txt"), new MockFileData(correctDataset));
+            _mockFileSystem.AddFile(Path.Combine(localDirFull, "local_incorrect.txt"), new MockFileData(textContents: "incorrect_data"));
+            _mockFileSystem.AddFile(Path.Combine(remoteDirFull, "remote_incorrect.txt"), new MockFileData(textContents: "incorrect_data"));
+
+            _rootConfig = new DataRootConfig(localDirFull, remoteDirFull);
+            _datasetLoader = new DatasetLoader(_rootConfig);
+        }
+
+        [Test]
+        public void ReturnsFromCorrectNameLocal()
+        {
+            Assert.IsNotNull(anObject: _datasetLoader.GetFromName(name: "local_correct.txt"),
+                message: "Loader did not manage to load local file.");
+        }
+
+        [Test]
+        public void ReturnsFromCorrectNameRemote()
+        {
+            Assert.IsNotNull(anObject: _datasetLoader.GetFromName(name: "remote_correct.txt"),
+                message: "Loader did not manage to load remote file.");
+        }
+
+        [Test]
+        public void ThrowsOnIncorrectName()
+        {
+            Assert.Throws<DatasetNotFoundException>(code: () => _datasetLoader.GetFromName(name: "invalid_name"),
+                message: "Loader accepted invalid file name.");
+        }
+
+        [Test]
+        public void ThrowsOnIncorrectFileContents()
+        {
+            Assert.Throws<DatasetFormatException>(code: () => _datasetLoader.GetFromName(name: "local_incorrect.txt"),
+                message: "Loader did not manage to load remote file.");
+        }
+
+        [Test]
+        public void DeletesIncorrectFilesFromLocal()
+        {
+            try
+            {
+                _datasetLoader.GetFromName(name: "remote_incorrect.txt");
+            }
+            catch (Exception)
```

Comment:
If it is ignored, we should listen for specific type of exception to not to hide errors in tests.